Alekseev

th4t1snik0 opened the pull request:
-HW3 DONE

th4t1snik0 committed:
-Создал класс Registry, в качестве примера взял два класса - Dog и Cat
- Добавил свой метод printAllItems()

th4t1snik0 committed:
-удалил out
-дореализовал методы класса Registry

lefasdayan commented on ru/turing/courses/lesson3/Registry.java:
```diff
@@ -1,13 +1,56 @@
-    private final Map<Object, Object> storage = new HashMap<>(); //todo поменять Object, Object на то, что будет у вашего класса в примере
+public class Registry<T, R> {
+    private final Map<T, R> storage = new HashMap<>(); //todo поменять Object, Object на то, что будет у вашего класса в примере
+    private final List<T> deletedKeys = new ArrayList<>();
```
Лист удаленных ключей лучше создавать внутри метода

lefasdayan commented on ru/turing/courses/lesson3/animal/Cat.java:
```diff
@@ -0,0 +1,78 @@
+    public void setWannaPlay(boolean wannaPlay) {
+        this.wannaPlay = wannaPlay;
+    }
+
```
Чтобы правильно работал метод equals() там, где он используется внутри регистра, нужно обязательно переопределить equals() и hashCode() у всех классов

lefasdayan commented on ru/turing/courses/lesson3/Example.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Example {
+    public static void main(String[] args) {
+        dogExample();
```
А зачем заводить отдельный метод и вызывать его в мэйне, если нужно просто весь код этого метода запустить из мэйна?

th4t1snik0 committed:
-List<> определяется в методе
-equals() и hashcode() переопределены во всех классах
-код вызывается из main()

lefasdayan commented on ru/turing/courses/lesson3/Example.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Аналогичный пример с котами
+     */
+    public static void catExample() {}
```
```suggestion

```

lefasdayan commented:
ЕЕЕ! Победа!
